docs/developer/taxes.mdx

```diff
@@ -394,3 +393,38 @@ mutation {
 ```
 
 For orders, pass the order ID as an argument and adjust the query to fetch the order data in the `taxableObject` field. To re-enable tax charging, pass `taxExemption: false`.
+
+## Address for tax calculation
+
+The process of determining the country code follows this sequence:
+
+1. Shipping address
+2. Billing address
+3. The channel's default country
+
+|             | Store Address | Shipping Address | Billing Address | Warehouse Address |
+| ----------- | ------------- | ---------------- | --------------- | ----------------- |
+| Country     | US            | US               | US              | US                |
+| CountryArea | LA            | VA               | TX              | VT                |
+
+### Example for shipable products
+
+- Create a checkout with channel and lines. Since the shipping address is empty, taxes are calculated based on the store address (Country US, state LA)
+- Update the billing address. The shipping address is still empty, but since the billing address is now available, taxes are calculated based on it (Country US, state TX)
+- Update the shipping address. Taxes are now calculated based on the shipping address (Country US, state VA).
+
+### Example for digital products
+
+- Create a checkout with channel and non-shippable lines. Since the shipping address is empty, taxes are calculated based on the store address (Country US, state LA)
+- Update the billing address. Taxes are calculated based on it (Country US, state TX)
+
+:::warning
+It's possible to set shipping address in the checkoutCreate mutation. In this case, the shipping address will be used for tax calculation since we always prioritize shipping address over billing address.
```

Comment:
Isn't it a bug that we don't clean up the shipping address when we don't have anything to send?
I would say it is a bug, as some mutations can raise `SHIPPING_NOT_REQUIRED` error.

Reply:
@korycins We change it in Saleor 3.15 https://github.com/saleor/saleor/blob/719d6c0da615161db8669a4626a90d6ebc45ccd3/CHANGELOG.md?plain=1#L350 


Until now, checkout mutations were ignoring provided shipping address when shipping was not required. After this change, the shipping address is always set when supplied in the input. It might be breaking, as the shipping address affects the country code used for tax calculation.
The order of resolving the checkout country code is always as follows:
Shipping address
Billing address
Channel's default country

Reply:
@szczecha I mean: You're writing, that in some cases (like checkoutCreate) we can set the shippingAddress, but later you will not be able to call `CheckoutShippingAddressUpdate`, as this mutation will raise an exception `SHIPPING_NOT_REQUIRED`. So it would be confusing for me, that in some cases, you can set shippingAddress, but with the same data, you will not be able to call another mutation, that will do the part of checkoutCreate work.

Reply:
I created an issue https://linear.app/saleor/issue/SHOPX-730/allow-to-set-shipping-address-for-digital-checkouts-in, when someone fixes it will also update the documentation